src/feathers/controls/supportClasses/LayoutViewPort.hx

```diff
@@ -396,4 +397,48 @@ class LayoutViewPort extends LayoutGroup implements IViewPort {
 			this._snapPositionsY = null;
 		}
 	}
+
+	private var _offscreenRendering:Bool = true;
+
+	/**
+		If the `offscreenRendering` value is `false`, all display objects offscreen the layout will automatically be set to visible false.
+
+		@since 1.0.0
+	**/
+	public var offscreenRendering(get, set):Bool;
+
+	private function get_offscreenRendering():Bool {
+		return this._offscreenRendering;
+	}
+
+	private function set_offscreenRendering(value:Bool):Bool {
+		if(value)
+		{
+			for (child in __children)
+			{
+				child.visible = true;
+			}
+		}
+
+		return this._offscreenRendering = value;
+	}
+
+	@:noCompletion private override function __enterFrame(deltaTime:Int):Void {
```

Comment:
I'm not happy hooking into OpenFL private/undocumented APIs, like `__enterFrame`. Can you make this part of the Feathers UI validation cycle instead? The visibility of children should need to be updated only when certain properties have changed. This can move into `update()`. Most likely, child visibility can be updated when the SCROLL, SIZE, or LAYOUT invalidation flag is set.

Reply:
> I'm not happy hooking into OpenFL private/undocumented APIs, like __enterFrame

I see that the CI failed because Flash doesn't have an `__enterFrame` method. That's another reason for changing the implementation.

Reply:
> I'm not happy hooking into OpenFL private/undocumented APIs, like `__enterFrame`. Can you make this part of the Feathers UI validation cycle instead? The visibility of children should need to be updated only when certain properties have changed. This can move into `update()`. Most likely, child visibility can be updated when the SCROLL, SIZE, or LAYOUT invalidation flag is set.

![image](https://user-images.githubusercontent.com/25794892/169197287-364e04fa-ac89-442a-9c98-2d32ff7509b3.png)

Unfortunately I can't because setInvalid doesn't work.

Reply:
Of course you can.

```haxe
if (this._manageChildVisibility || (this._currentLayout is IScrollLayout)) {
```